gdal/alg/contour.cpp

```diff
@@ -58,8 +58,11 @@ static CPLErr OGRPolygonContourWriter( double dfLevel,
     if( poInfo->nIDField != -1 )
         OGR_F_SetFieldInteger( hFeat, poInfo->nIDField, poInfo->nNextID++ );
 
-    if( poInfo->nElevField != -1 )
-        OGR_F_SetFieldDouble( hFeat, poInfo->nElevField, dfLevel );
+    if( poInfo->nElevFieldMin != -1 )
+        OGR_F_SetFieldDouble( hFeat, poInfo->nElevFieldMin, dfLevelMin );
```

Comment:
which value is written for the polygon at lowest elevation ? -inf ? in that case if would be better not to write it at all as some formats might have issues with infinite values

Reply:
Yes it's -Inf to be consistent with +Inf for elevation greater than the last specified level. 

If we remove the polygon lower than the first specified level, I assume we have to remove the polygon after the last specified level. It means that user has to know its min/max raster value if he wants the contours to cover the whole raster.

Or maybe we could set min/max raster value as limits instead of -Inf/+Inf.

Reply:
I didn't mean not to write the polygon, but just not to write an infinite value for an attribute: it is fine to let a field empty. That, or setting the min/max raster value would be fine too.

Reply:
OK, so I choose to set raster min/max as limit contours. See my last commit